packages/core/src/lib/jdk/JdkHelper.ts

```diff
@@ -56,29 +58,58 @@ export class JdkHelper {
    */
   async runJava(args: string[]): Promise<{stdout: string; stderr: string}> {
     const java = this.process.platform === 'win32' ? '/bin/java.exe' : '/bin/java';
-    const runJavaCmd = this.joinPath(this.getJavaHome(), java);
+    const runJavaCmd = this.joinPath(this.getJavaHome(this.config.jdkPath, this.process),
+        java);
     return await executeFile(runJavaCmd, args, this.getEnv());
   }
 
   /**
    * Returns information from the JAVA_HOME, based on the config and platform.
-   * @returns {string} the value for the JAVA_HOME
+   * @param {Config} config The bubblewrap general configuration
+   * @param {NodeJS.Process} process Information from the OS process
    */
-  getJavaHome(): string {
-    if (this.process.platform === 'darwin') {
-      return this.joinPath(this.config.jdkPath, '/Contents/Home/');
-    } else if (this.process.platform === 'linux' || this.process.platform === 'win32') {
-      return this.joinPath(this.config.jdkPath, '/');
+  getJavaHome(jdkPath: string, process: NodeJS.Process): string {
```

Comment:
The old version of the method doesn't have any parameters and uses the jdkPath / process passed as parameter to the config. We want to keep the old signature.